flink-connectors/flink-connector-aws-kinesis-data-streams/src/test/java/org/apache/flink/connectors/kinesis/testutils/KinesaliteContainer.java

```diff
@@ -99,25 +99,41 @@ public Properties getHostProperties() {
     }
 
     /** Returns the client to access the container from outside the docker network. */
-    public KinesisAsyncClient getContainerClient() throws URISyntaxException {
-        return getClient(getContainerEndpointUrl());
+    public KinesisAsyncClient getContainerClient(SdkAsyncHttpClient httpClient)
+            throws URISyntaxException {
+        return getClient(getContainerEndpointUrl(), httpClient);
     }
 
-    /** Returns the client to access the host from inside the docker network. */
+    /** Returns the client with the default async http client. */
     public KinesisAsyncClient getHostClient() throws URISyntaxException {
-        return getClient(getHostEndpointUrl());
+        return getHostClient(buildSdkAsyncHttpClient());
```

Comment:
How you intend to close the http client? From what I've seen with firehose the wrapping client doesn't seem to close the underlying resource.

Reply:
`@deprecated` this one in case anyone is using it from the test jar

Reply:
@CrynetLogistics deprecated what where? I cannot see this change in this PR. We do not need to maintain backwards compatibility in our test code. Are you worried if external code is relying on Flink test-jars?

Reply:
I've removed it instead now.